APCAppCore/APCAppCore/UI/TabBarControllers/Profile/APCDownloadDataViewController.m

```diff
@@ -0,0 +1,187 @@
+        NSDate *today = [NSDate date];
+        _startDatePicker.date = today.startOfYear;
+        [_startDatePicker setMaximumDate:today];
+        _startDatePicker.datePickerMode = UIDatePickerModeDate;
+    }
+    
+    return _startDatePicker;
+}
+
+- (UIDatePicker *) endDatePicker {
+    if (!_endDatePicker) {
+        _endDatePicker = [[UIDatePicker alloc] init];
+        NSDate *today = [NSDate date];
+        _endDatePicker.date = today;
+        [_endDatePicker setMaximumDate:today];
+        _endDatePicker.datePickerMode = UIDatePickerModeDate;
+    }
+    
+    return _endDatePicker;
+}
+
+- (APCUser *) user {
+    if (!_user) {
+        _user = ((APCAppDelegate *)[UIApplication sharedApplication].delegate).dataSubstrate.currentUser;
+    }
+    
+    return _user;
+}
+
+#pragma mark - Actions
+
+-(void)updateDates:(id) __unused sender
+{
+    [self.startTextField resignFirstResponder];
+    [self.endTextField resignFirstResponder];
+    [self updateDateTextFields];
+}
+
+-(void)updateDateTextFields
+{
+    NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+    dateFormatter.timeStyle = NSDateFormatterNoStyle;
+    dateFormatter.dateStyle = NSDateFormatterMediumStyle;
+    self.startTextField.text = [dateFormatter stringFromDate:self.startDatePicker.date];
+    self.endTextField.text = [dateFormatter stringFromDate:self.endDatePicker.date];
+}
+
+- (IBAction)downloadData:(id) __unused sender {
+    self.user.downloadDataStartDate = self.startDatePicker.date;
+    self.user.downloadDataEndDate = self.endDatePicker.date;
+    
+    APCSpinnerViewController *spinnerController = [[APCSpinnerViewController alloc] init];
+    [self presentViewController:spinnerController animated:YES completion:nil];
+    
+    typeof(self) __weak weakSelf = self;
+    [self.user sendDownloadDataOnCompletion:nil];
+    
+    
+    [self.user sendDownloadDataOnCompletion:^(NSError *error) {
+        if (error) {
+            
+            APCLogError2 (error);
+            
+            [spinnerController dismissViewControllerAnimated:NO completion:^{
+                
+                UIAlertAction *cancelAction = [UIAlertAction actionWithTitle:NSLocalizedString(@"Dismiss", @"Dismiss") style:UIAlertActionStyleCancel handler:^(UIAlertAction * __unused action) {
+                    
+                }];
+                
+                UIAlertController *alertController = [UIAlertController alertControllerWithTitle:@"" message:error.message preferredStyle:UIAlertControllerStyleAlert];
+                [alertController addAction:cancelAction];
+                
+                [weakSelf.navigationController presentViewController:alertController animated:YES completion:nil];
+            }];
+        }
+        else
+        {
+            [spinnerController dismissViewControllerAnimated:NO completion:^{
+                
+                [weakSelf dismissViewControllerAnimated:YES completion:nil];
```

Comment:
How hard is it to add an interstitial screen here that looks-ish like the one you get when you finish a task and says "Your data will be emailed to you shortly" or similar, with a done button to go back? Otherwise I could see users wondering whether it actually did anything... but if it's a lot of effort to add, we can try without and fix later if need be.